test/e2e/cluster_upgrade_runtimesdk.go

```diff
@@ -484,3 +509,28 @@ func clusterConditionShowsHookFailed(cluster *clusterv1.Cluster, hookName string
 		strings.Contains(conditions.GetMessage(cluster, clusterv1.TopologyReconciledCondition), hookFailedMessage) &&
 		strings.Contains(conditions.GetMessage(cluster, clusterv1.TopologyReconciledCondition), hookName)
 }
+
+func backupAndDeleteCluster(ctx context.Context, proxy framework.ClusterProxy, namespace, clusterName, artifactFolder string) {
+	By("Deleting the workload cluster")
+	cluster := &clusterv1.Cluster{}
+	Eventually(func() error {
+		return proxy.GetClient().Get(ctx, client.ObjectKey{Namespace: namespace, Name: clusterName}, cluster)
+	}).Should(Succeed())
+
+	// Dump all the logs from the workload cluster before deleting them.
+	proxy.CollectWorkloadClusterLogs(ctx, cluster.Namespace, cluster.Name, filepath.Join(artifactFolder, "clusters-beforeClusterDelete", cluster.Name))
+
+	// Dump all Cluster API related resources to artifacts before deleting them.
+	framework.DumpAllResources(ctx, framework.DumpAllResourcesInput{
+		Lister:    proxy.GetClient(),
+		Namespace: namespace,
+		LogPath:   filepath.Join(artifactFolder, "clusters-beforeClusterDelete", proxy.GetName(), "resources"),
+	})
+
+	By("Deleting the workload cluster")
+	// Delete the workload cluster and call the before ClusterDeleteTestHandler.
```

Comment:
```suggestion
```
It's now only delete. I think the call is pretty obvious + we have the log directly above in l.530. Let's just drop the comment